graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/federation/FederatedSchemaGenerator.kt

```diff
@@ -40,24 +40,15 @@ open class FederatedSchemaGenerator(generatorConfig: FederatedSchemaGeneratorCon
     }
 
     /**
-     * Scans specified packages for all the federated (extended) types and adds them to the target schema.
+     * Scans specified packages for all the federated (extended) types and adds them to the target schema before generating the rest of the schema
      */
     @Suppress("Detekt.SpreadOperator")
     fun GraphQLSchema.Builder.federation(supportedPackages: List<String>): GraphQLSchema.Builder {
         val scanResult = ClassGraph().enableAllInfo().whitelistPackages(*supportedPackages.toTypedArray()).scan()
 
         scanResult.getClassesWithAnnotation(ExtendsDirective::class.jvmName)
             .map { it.loadClass().kotlin }
-            .map {
-                val graphQLType = if (it.isAbstract) {
-                    interfaceType(it)
-                } else {
-                    objectType(it)
-                }
-
-                // workaround to explicitly apply validation
-                config.hooks.didGenerateGraphQLType(it.createType(), graphQLType)
```

Comment:
workaround is no longer needed?

Reply:
Correct, as we now call `grahpQLTypeOf` that triggers all the proper hooks